### ISSUE TITLE
[React Icon Tooltip] carry over a11y enhancements from Vanilla

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3811]
A11y changes where made to the HTML structure in Icon Tooltip in this PR. These changes should be applied to the React version

https://github.com/carbon-design-system/carbon/pull/3115
